packages/wonder-blocks-dropdown/components/single-select-menu.js

```diff
@@ -0,0 +1,168 @@
+import SelectItem from "./select-item.js";
+
+export type SingleSelectItemProps = {
+    /** Whether this item is disabled. Default false. */
+    disabled?: boolean,
+    /** Display text of the item. */
+    label: string,
+    /** Value of this item. Treat as a key. */
+    value: string,
+    /** Optional extra callback. Passes whether this item is selected. */
+    onClick?: (selected: boolean) => void,
+};
+
+type Props = {
+    /**
+     * The items in this menu.
+     */
+    items: Array<SingleSelectItemProps>,
+
+    /**
+     * Callback for when the selection of the menu changes. Parameter is the
+     * newly selected item.
+     */
+    onChange: (selectedValue: string) => void,
+
+    /**
+     * Placeholder for the opening component when there are no items selected.
+     */
+    placeholder: string,
+
+    /**
+     * Value of the currently selected item for this menu.
+     */
+    selectedValue?: string,
+
+    /**
+     * Whether this menu should be left-aligned or right-aligned with the
+     * opener component. Defaults to left-aligned.
+     */
+    alignment?: "left" | "right",
+
+    /**
+     * Whether to display the "light" version of this component instead, for
+     * use when the item is used on a dark background.
+     */
+    light?: boolean,
+
+    /**
+     * Whether this menu is disabled. A disabled menu may not be opened and
+     * does not support interaction. Defaults to false.
+     */
+    disabled?: boolean,
+
+    /**
+     * Optional styling to add to dropdown.
+     */
+    style?: any,
+};
+
+type State = {
+    /**
+     * Whether or not menu is open.
+     */
+    open: boolean,
+};
+
+export default class SingleSelectMenu extends React.Component<Props, State> {
+    static defaultProps = {
+        alignment: "left",
+        disabled: false,
+        light: false,
+    };
+
+    constructor(props: Props) {
+        super(props);
+
+        this.state = {
+            open: false,
+        };
+    }
+
+    toggleMenu() {
+        this.setState((prevState) => ({
+            open: !prevState.open,
+        }));
+    }
+
+    handleSelected(selectedValue: string) {
+        this.setState({
+            open: false, // close the menu upon selection
+        });
+
+        // Call callback if selection changes.
+        if (selectedValue !== this.props.selectedValue) {
+            this.props.onChange(selectedValue);
+        }
+    }
+
+    render() {
+        const {
+            alignment,
+            disabled,
+            items,
+            light,
+            placeholder,
+            selectedValue,
+            style,
+        } = this.props;
+
+        const {open} = this.state;
+
+        const selectedItem = items.find((item) => item.value === selectedValue);
+        // If nothing is selected, or if the selectedValue doesn't match any
+        // item in the menu, use the placeholder.
+        const menuText = `${selectedItem ? selectedItem.label : placeholder}`;
```

Comment:
The template string isn't necessary b/c `placeholder` and `selectedItem.label` are already strings.